### DISCUSSION TITLE
How to validate URL?

### DISCUSSION THREAD

**Discussion by ghost** [directus/directus#5365] [Archive]
I’m new to directus and I couldn’t figure out how to write custom validations for field inputs yet? I don’t find anything in the docs … that’s a bit frustrating … 

How would you validate a simple field like an URL field?

Can anyone help?

**Comment by rijkvanzanten** [comment]
You can setup Validation under permissions (https://docs.directus.io/concepts/permissions/#permissions). For URL specific, you could rely on the regex filter operator: https://docs.directus.io/reference/filter-rules/#filter-operators

**Reply by ghost** [reply]
Thanks for your hint!
Oh wow! That is complicated.
Is there a way to always use the same validation rules for create- and update-permissions?
And what if I want to apply my validation rules to all roles? Do I have to set them on every roles permission?

**Reply by ghost** [reply]
Why does this simple regex not work inside directus (field validation)?

```
{
    "url": {
        "_regex": "^https?://\S*"
    }
}
```

A red cross appears and the regex could not be saved. If I remove the backslash, it will save but the regex is of course wrong.

**Reply by rijkvanzanten** [reply]
You'll have to escape that backslash 👍🏻 `^https?://\\S*`
